hack/tools/logcheck/main.go

```diff
@@ -179,11 +179,16 @@ func isKeysValid(keyValues []ast.Expr, fun ast.Expr, pass *analysis.Pass, funNam
 	}
 }
 
-func checkForFormatSpecifier(expr *ast.CallExpr, pass *analysis.Pass) bool {
+func checkForFormatSpecifier(expr *ast.CallExpr, pass *analysis.Pass, c *config) bool {
 	if selExpr, ok := expr.Fun.(*ast.SelectorExpr); ok {
 		// extracting function Name like Infof
 		fName := selExpr.Sel.Name
-		if specifier, found := hasFormatSpecifier(expr.Args); found {
+		specifier, found := hasFormatSpecifier(expr.Args)
+		if found {
+			if (fName == "Exit" || fName ==  "Exitf" || fName == "Exitln" || fName == "ExitDepth") && c.allowUnstructured {
```

Comment:
Looks like your implementation was meant to allow always allow Exit* functions, but this is not what I intended.
We should treat Exit* functions same as other unstructured methods. Meaning that the fix should just update isUnstructured function to include Exit* functions.

Reply:
i understand, i will modify it immediately